### PR TITLE
webui(dm): add many fix of webui

### LINKED ISSUE #4956: tracking issue for bugfix of webui first release

### LINKED ISSUE #4993: DM WebUI need to remove dm-worker offline option
### What did you do?

- deploy dm cluster，create source
- webui-> Migration->Cluster->members->workers, try to click offline (not work)

I suggest to remove this click on the WebUi cause it's not the correct function.

### What did you expect to see?

worker can offline

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
nightly
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #4956  close #4993

### What is changed and how it works?

* fix convert between taskCfg and openapi.Task in when only route/filter schema
* add many ui changes

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by baurine — APPROVED** [review]
lgtm

**Review by D3Hunter — COMMENTED** [review]
didn't review code about ts scripts, others lgtm

**Review by buchuitoudegou — APPROVED** [review]
Not severe issue. LGTM

File: dm/dm/config/task_converters.go [hunk 1]
```diff
@@ -542,26 +560,35 @@ func SubTaskConfigsToOpenAPITask(subTaskConfigList []*SubTaskConfig) *openapi.Ta
 			}
 			tableMigrateRule.BinlogFilterRule = &ruleNameList
 		}
+		ruleKey := fmt.Sprintf("%s-%s-%s", sourceName, schemaPattern, tablePattern)
```
[the remark this answers is not in the document]
Comment (Ehco1996) [reply dm/dm/config/task_converters.go]: addressed but can you tell me why `strings.Join` is better?  Is it better performance?

[the remark this answers is not in the document]
Comment (D3Hunter) [reply dm/dm/config/task_converters.go]: simpler to me, ok not to change

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**